datrie/tail.c

```diff
@@ -251,6 +251,57 @@ tail_fwrite (const Tail *t, FILE *file)
 }
 
 
+size_t
+get_tail_serialized_size (const Tail *t)
+{
+    size_t static_count = (
+        sizeof(int32) * 3 // TAIL_SIGNATURE , t->first_free , t->num_tails
+    );
+    size_t dynamic_count = 0u;
+    if(t->num_tails > 0){
+        dynamic_count += (
+            4 * 2 // t->tails[i].next_free , t->tails[i].data
+            + 2 // length
+        ) * t->num_tails;
+        for (TrieIndex   i = 0; i < t->num_tails; i++) {
+            if (t->tails[i].suffix > 0)
+            {
+                dynamic_count += strlen ((const char *)t->tails[i].suffix) * sizeof(const char);
+            }
+        }
+    }
+    return static_count + dynamic_count;
+}
+
+
+int
+tail_serialize (const Tail *t, unsigned char **ptr)
+{
+    TrieIndex   i;
+
+    serialize_int32_be_incr (ptr, TAIL_SIGNATURE);
+    serialize_int32_be_incr (ptr, t->first_free);
+    serialize_int32_be_incr (ptr, t->num_tails);
+
+    for (i = 0; i < t->num_tails; i++) {
+        int16   length;
+        serialize_int32_be_incr (ptr, t->tails[i].next_free);
+        serialize_int32_be_incr (ptr, t->tails[i].data);
+
+        length = t->tails[i].suffix ? strlen ((const char *)t->tails[i].suffix)
+                                    : 0;
+        serialize_int16_be_incr (ptr, length);
+        if (length)
+        {
+            memcpy (*ptr, (char *)t->tails[i].suffix, length);
+            *ptr += length;
```

Comment:
`length` -> `length * sizeof (TrieChar)` in both lines, to be consistent with the size calculation?

Reply:
No, it is not the same. There we were counting size in bytes, here we use not `void *` but `TrieChar *`, so the size is in `TrieChar`s.